app/models/email_template.rb

```diff
@@ -0,0 +1,58 @@
+class EmailTemplate < ActiveRecord::Base
+
+  enum kind: [:user_welcome, :ticket_received]
+
+  scope :by_kind, -> (k) { where(kind: kinds[k]) }
+  scope :active, -> { where.not(draft: true) }
+
+  validates_presence_of :kind
+
+  def self.create_default_templates(kinds, options={})
+    # default templates are always english
+    kinds.each do |kind|
+     create prefill_by(kind).merge(options)
+    end
+  end
+
+  def user_welcome_kind?
+    kind == 'user_welcome'
+  end
+
+  def ticket_received_kind?
+    kind == 'ticket_received'
+  end
+
+  def custom_kind?
+    kind == 'custom'
+  end
```

Comment:
By defining an enum, you automatically get functions suchs as `user_welcome?` and `ticket_received?`, so your custom functions are not needed.